sql/src/main/java/io/crate/expression/scalar/string/TrimFunction.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.primitives.Chars;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.expression.scalar.TripleScalar;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionInfo;
+import io.crate.types.DataTypes;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashSet;
+
+public final class TrimFunction {
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(new TripleScalar<>(functionInfo(), TrimFunction::trim));
+    }
+
+    private static FunctionInfo functionInfo() {
+        return new FunctionInfo(
+            new FunctionIdent("trm", ImmutableList.of(DataTypes.STRING, DataTypes.STRING, DataTypes.STRING)),
+            DataTypes.STRING);
+    }
+
+    @Nullable
+    private static String trim(@Nonnull String mode,
+                               @Nullable String charsToTrim,
+                               @Nullable String value) {
+        if (value == null || charsToTrim == null) {
+            return value;
+        }
+
+        final HashSet<Character> trimmableChars = new HashSet<>(Chars.asList(charsToTrim.toCharArray()));
+        int start = 0;
+        int len = value.length();
+
+        if (isLeadingOrBothMode(mode)) {
+            while (start < len && trimmableChars.contains(value.charAt(start))) {
+                start++;
+            }
+        }
+
+        if (isTrailingOrBothMode(mode)) {
+            while (start < len && trimmableChars.contains(value.charAt(len - 1))) {
+                len--;
+            }
+        }
+
+        return value.substring(start, len);
+    }
+
+    private static boolean isLeadingOrBothMode(@Nonnull String mode) {
+        return mode.equals("LEADING") || mode.equals("BOTH");
```

Comment:
Could we make this more strict somehow? E.g. introduce an enum and parse it once - with a failure if it is neither of the 3 allowed values.